src/Response/Serializer.php

```diff
@@ -106,6 +106,6 @@ private static function getStatusLine(StreamInterface $stream)
             throw new UnexpectedValueException('No status line detected');
         }
 
-        return [$matches['version'], $matches['status'], isset($matches['reason']) ? $matches['reason'] : ''];
+        return [$matches['version'], (int) $matches['status'], isset($matches['reason']) ? $matches['reason'] : ''];
```

Comment:
This is exactly the point I was making previously; with the change you suggest above, we now have a BC break requiring consumers make changes such as this.